bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java

```diff
@@ -180,7 +195,47 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 device.setMute(MuteInstructionChannel.VIDEO, muteOn);
               }
             }
+          } else {
+            logger.debug("Received unknown audio/video mute command {}", command);
+          }
+          break;
+        case CHANNEL_TYPE_LAMP_ACTIVE:
+        case CHANNEL_TYPE_LAMP_HOURS:
+          if (command == RefreshType.REFRESH) {
+            List<LampState> lampStates = device.getLampStatesCached();
+            // update all lamp related channels, as the response contains information about all of them
+            for (Channel lampChannel : thing.getChannels()) {
+              String lampChannelTypeId = getChannelTypeId(lampChannel.getUID());
+              if (lampChannelTypeId == null) {
+                continue;
+              }
+              boolean isLampActiveChannel = CHANNEL_TYPE_LAMP_ACTIVE.equals(lampChannelTypeId);
+              boolean isLampHoursChannel = CHANNEL_TYPE_LAMP_HOURS.equals(lampChannelTypeId);
+
+              if (isLampActiveChannel || isLampHoursChannel) {
+                int lampNumber = ((BigDecimal) lampChannel.getConfiguration().get(CHANNEL_PARAMETER_LAMP_NUMBER))
+                    .intValue();
+                try {
+                  LampState lampState = lampStates.get(lampNumber - 1);
+                  if (isLampActiveChannel) {
+                    updateState(lampChannel.getUID(), lampState.isActive() ? OnOffType.ON : OnOffType.OFF);
+                  }
+                  if (isLampHoursChannel) {
+                    updateState(lampChannel.getUID(), new DecimalType(lampState.getLampHours()));
+                  }
+                } catch (IndexOutOfBoundsException e) {
+                  logger.debug("Status information for lamp {} is not available", lampNumber);
+                  throw new ConfigurationException("Status information for lamp " + lampNumber + " is not available");
```

Comment:
Is such exception appropriate for handleCommand method ?

Reply:
This exception is caught in the same method and handled using handleConfigurationException(e), which sets the device offline and sets an appropriate error text.